### ISSUE TITLE
Groovy formatter with custom config file skips most of the rules

### ISSUE THREAD

**Issue by abialas** [diffplug/spotless#431]
I am trying to enable groovy formatter with custom xml file (which contains rules exported from the eclipse). However, most of the rules are not applied.

- Gradle version: 5.1
- Spotless version: I tried with 3.4.0 and 3.24.1
- Microsoft Windows 7 Professional
- Spotless configuration:

`    spotless {
        groovy {
            greclipse().configFile("$rootDir/buildSrc/formatter.xml")
        }
    }`

No errors are in logs.

**Comment by nedtwigg** [comment]
What's the content of your `formatter.xml`?

**Comment by fvgh** [comment]
@abialas 
I hope you followed more or less [my documentation](https://github.com/diffplug/spotless/tree/master/plugin-gradle#groovy-eclipse). So the formatter.xml contains your [exported Java formatter properties](https://github.com/diffplug/spotless/blob/master/ECLIPSE_SCREENSHOTS.md).
As you see in the docs, there are [additional poperties](https://github.com/diffplug/spotless/blob/master/testlib/src/main/resources/groovy/greclipse/format/greclipse.properties) which are groovy specific.

I recommends to do the configuration within Eclipse, using the Groovy-Eclipse plugin to see all supported configuration options. Ass described in the docs, the properties are saved in `org.codehaus.groovy.eclipse.ui.prefs`, which you can find in a sub-directory of your Eclipse workspace.

When you use this approach, you directly see the Groovy-Eclipse documentation, wich states that only tabs and space related preferences are inherited from the Java formatter.

I will enhance the documentation to emphasize that only some properties are taken from the Java formatter.

**Comment by fvgh** [comment]
BTW: I am afraid that also the link in the docs for the additional properties is outdated and requires an update.

**Comment by abialas** [comment]
In Groovy-Eclipse formatter settings it is written that: "Tab and space related preferences are inherited from the Java Formatter". I have this setting in Java formatter:

`<setting id="org.eclipse.jdt.core.formatter.insert_space_before_opening_paren_in_if" value="do not insert"/>`

but this does not format my code (does not enter space after if and the opening bracket).

**Comment by fvgh** [comment]
Sorry @abialas , I am afraid this is just your wishful interpretation of the Groovy-Eclipse documentation.

The [additional properties](https://github.com/diffplug/spotless/blob/master/testlib/src/main/resources/groovy/greclipse/format/greclipse.properties) I linked in my previous comment were, as far as I remember, the only configuration options, when I provided the first Spotless interface for the Groovy-Eclipse formatter.

I only provide an interface to use Eclipse formatters as a standalone function and assuring (as far as possible, there are conceptual limits regarding the AST) that formatting triggered by Spotless leads to the same results as within Eclipse IDE.

Note that Groovy-Eclipse team asked in some of their issues for help regarding their formatter implementation.
